github release management tools

jam7 opened the pull request:
I've made two scripts to make binary release page maintenance easy.

It is discussed at #268.

I made test page at https://github.com/jam7/chromebrew/releases which contains all binary files refered by newgcc branch.  This page is made by following steps with two newly created scripts.  Isn't it easy?  :)
```
<<create release page first and give it a TAG>>
$ sh create_url_list.sh -s > url_list
$ wget -P bin -i url_list
$ sh upload_github_release.sh TAG bin/*
```

jam7 commented:
FYI, I use github-release at https://github.com/aktau/github-release, so golang is required to use this.  It is written in comment of script files.

uberhacker commented:
This is awesome @jam7!  Were you thinking of a possible "official" binary repo location or did you just want to make it easy for folks to create their own "mirror" repo?

jam7 commented:
@uberhacker Thanks.  I made this to help skycocker and collaborators to make their "official" binary repo and maintain it easily at http://github.com/skycocker/chromebrew/release.

Regarding to "mirror" repo stuff, I didn't think about it.  create_url_list.sh will help to make a list of URL needed to be mirrored.

jam7 commented:
I'll explain little about the scenarios to use these scripts.  I think only skycocker and collaborators can performs this at here, https://github.com/skycocker/chromebrew/.

## setup

Need to install several packages and set GITHUB_TOKEN environment variable.  Check https://github.com/jam7/simple-github-release#prerequirements and https://github.com/jam7/simple-github-release#setup.

GITHUB_OWNER and GITHUB_REPO is not needed since it is hard coded in these scripts.

## starting "official" release page with binary files

1. create a release page first.  need to remeber tag you made here
2. need to download all binaries using
```
$ git clone git@github.com:skycocker/chromebrew.git
$ cd chromebrew
$ sh create_url_list.sh -s > url_list
$ wget -P bin_dir -i url_list
```
3. upload them all
```
$ sh upload_github_release.sh TAG-YOU-SPECIFY bin_dir/*
```
4. edit all package files to refer new released binary and re-tag

## merging PR with binary url

1. create a pre-release page.  need to remeber tag you made here
2. need to download new binaries (not from github.com) using
```
$ git clone ...
$ cd chromebrew
$ sh create_url_list.sh -s > url_list
$ wget -P bin_dir -i url_list
```
3. upload them all.  edit package files need to be updated and re-tag

## next release

There are two scenario.  If you want to put only updated binaries there, just edit pre-release page.

If you want to put all binaries each time, performs above starting scenario again with following command at 2nd step.  `-a` shows all binary URLs including github.com.
```
$ sh create_url_list.sh -a > url_list
```

minektur commented:
I'm going to have to try this out on a test repo of my own.  I have no idea if you already do this, but it would be great if the script would stash a version into the crew 'binary' , tag everything, and then make a realease based on that tag and the current available binaries.  

As an aside we would also want to have people submit the version they are running when they submit bugs... anyway - not having yet tested this - I think this is a move in the right direction.

jam7 commented:
@minektur thanks.  please try it.  please change owner to you at the very beginning of upload script for the test.

thedamian commented:
looks good. 

I want a couple of tests then I'm ready to merge because this looks great!

jam7 commented:
I've noticed install.sh may contain different URL from ruby.rb.  So, I modified create_url_list.sh script to check both install.sh and packages/*.rb files.  I also implemented a script to make a list of sha1/sha256 in both install.sh and packages/*.rb.  These help maintenance of binary packeages.

thedamian commented:
Sorry I haven't merged this. New files. so no problem.
